pandas/io/xml.py

```diff
@@ -98,17 +103,17 @@ class _XMLFrameParser:
 
     def __init__(
         self,
-        path_or_buffer,
-        xpath,
-        namespaces,
-        elems_only,
-        attrs_only,
-        names,
-        encoding,
-        stylesheet,
+        path_or_buffer: FilePath | ReadBuffer[bytes] | ReadBuffer[str],
+        xpath: str,
+        namespaces: dict | None,
+        elems_only: bool,
+        attrs_only: bool,
+        names: Sequence[str] | None,
```

Comment:
I think in other places, we assume that column names just have to be `Hashable`.

Reply:
Good point, forgot that this is not restricted by the xml spec

Reply:
Do you have any idea why this causes

```
 error: Key expression in dictionary comprehension has incompatible type "Hashable"; expected type "str"  [misc]
```
when changing to Hashable? This smells like a mypy bug?

Reply:
Ok missed that the variable was typed, but we use functions which expect a string on the names, so the annotation was correct